---
title: "220807_Soil_pH_examine_new_pipeline"
author: "Kiseok Lee"
date: "2022-08-07"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


```{r global_options, include=FALSE}
library(knitr)
knitr::opts_chunk$set(fig.width=11, fig.height=9,
                      error=TRUE, echo=TRUE, eval=TRUE, warning=FALSE, message=FALSE,
                      tidy.opts=list(width.cutoff=40),tidy=TRUE)
```

## pH perturabtion and adaptation - examining new pipeline.
Researcher & Analysis: **Kiseok Lee** \
Experiment Date: 6/10/22 - 6/17/22 (5 day each) \
Analysis Date: 8/7/22 \
Lab: **Seppe Kuehn** at UChicago \

## Import libraries and setup
```{r}
# libraries
library(phyloseq)
library(dplyr)
library(ggplot2)
library(RColorBrewer)
library(vegan)
library(tidyverse)
library(magrittr)
library(readxl)
library(reshape2)
library(gtools)
library(devtools)
library(openxlsx)
library(ape)
library(stringr)
library(tidyr)
library(ggrepel)
library(ggpubr)
library(scales)
require(gridExtra)
library(grid)
library(agricolae)

library(BiocManager)


# do I need these?
#library(seqtime)
# BiocManager::install("seqtime")
library(metagenomeSeq)
# BiocManager::install("metagenomeSeq", version = "3.12")


# color collection
my_color_collection <- c(
  "#CBD588", "#5F7FC7", "orange", "#AD6F3B", "#673770", 
  "#D14285", "#652926", "#C84248", "#8569D5", "#5E738F",
  "#D1A33D", "#8A7C64", "#599861","#616163", "#FFCDB2",
  "#6D9F71", "#242F40",
  "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724")

my_color_collection2 <- c(
  "gray",'black',"#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  '#E55812',  "#FFCDB2", "#242F40", "#6D9F71", "#CCA43B", 
  "#F92A82", "#ED7B84", "#5F7FC7", 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
  "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
  "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724"
)

my_color_300 <- c(
  "black","#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
   "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
   "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
   "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
   "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
   "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange",  
  "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861",
  "#616163",  "#FFCDB2", "#242F40", "#6D9F71", 
   "#CCA43B", "#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  '#9FB1BC', '#D3D0CB', '#E2C044', '#5BC0EB', '#FDE74C', 
  '#9BC53D', '#E55934', '#FA7921', "#CD9BCD", "#508578", "#DA5724",
  "#CBD588", "#5F7FC7", "orange")

## theme for ggplot
mytheme <- theme_bw() + 
  theme(text = element_text(face="bold", colour = 'black'),
        plot.title = element_text(size = 19,hjust = 0.5,face="bold", colour = 'black'),
        axis.title.x = element_text(size = 17,hjust = 0.5,face="bold", colour = 'black'),
        axis.title.y = element_text(size = 17,hjust = 0.5,face="bold", colour = 'black'),
        axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13,face="bold", colour = 'black'),
        axis.text.y = element_text(size=13,face="bold", colour = 'black'),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.background=element_blank(),panel.border=element_blank(),plot.background=element_blank(),
        axis.ticks = element_line(size = 1.1),
        legend.text=element_text(size=10,face="bold", colour = 'black'))

mytheme_2d <- theme_bw() +
  theme(text = element_text(face="bold", colour = 'black'),
        plot.title = element_text(size = 19,hjust = 0.5,face="bold", colour = 'black'),
        axis.title.x = element_text(size = 17,hjust = 0.5,face="bold", colour = 'black'),
        axis.title.y = element_text(size = 17,hjust = 0.5,face="bold", colour = 'black'),
        axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13,face="bold", colour = 'black'),
        axis.text.y = element_text(size=13,face="bold", colour = 'black'),
        axis.ticks = element_line(size = 1.1),
        legend.text=element_text(size=10,face="bold", colour = 'black'),
        panel.grid.major = element_blank(),
        panel.background=element_blank(),plot.background=element_blank()
        )
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank(),panel.background=element_blank(),plot.background=element_blank())


# for git push, use this instead of using wflow_git_push()
# git push -u origin master (in the Git app / in the working directory)

# for making pdf file
# library(rmarkdown)
# render("analysis/~~.Rmd", "pdf_document")

```


```{r, echo=F}

###### phyloseq microbiome package #######
### 0. Data manipulation practice
# otu_table(phy)
# tax_table(phy)
# meta(phy)  # same as this sample_data(phy)
# psmelt #important
# psmelt?? otu table?????? ?Ǵ? ???? ?ƴ϶?, phyloseq ??ü???? ?? ?ȴ?.
# phyloseq_summary(phy)
# phy.bac <- subset_taxa(phy, Phylum == "D_1__Chloroflexi")
# sample_variables(phy)
# sample_names(phy)
# phy.subset <- subset_samples(phy, LargeScaleGeo == "CC")
# f <- filter_taxa(phy, function(x) var(x) > 1e-05, TRUE)
# s <- sample_sums(phy)
# taxa_names(phy.PAB)  # get otus
# ## Remove residual taxa that don??t have any sequences
# sum(taxa_sums(phy.clean.20) == 0)
# taxa_sums(phy.clean)
# taxa_sums(phy.clean.20)
# phy.clean.filt <- filter_taxa(phy.clean.20, function(x) sum(x) != 0, TRUE)
# sample_sums
# taxa_sums
# m.map <-import_qiime_sample_data("merge_metadata.tsv")
# SD = merge_samples(sample_data(GP), "SampleType")
# transform_sample_counts()
# ps3ra = transform_sample_counts(ps3, function(x){x / sum(x)})
# phy.m <-  merge_samples(phy.clean.ss.5, "Label")
# phy.clean.log2 <- transform_sample_counts(phy.clean.nolog, function(x){log2(1+x)})
# phy.clean.log10 <- transform_sample_counts(phy.clean.nolog, function(x){log(1+x)})
# df.node$label <- ifelse(grepl("^B",df.node$id),'Bacteria','Fungi')
# rownames(df.phy.css.ss.5) <- phy.vec$OTU_id[match(rownames(df.phy.css.ss.5),phy.vec$OTU)]

```


## Import biom file
```{r}

### We can then load the biom file with phyloseq function import_biom. We extract the OTU table with OTU abundances and the taxonomy table from the resulting phyloseq object.
bac_phylo=import_biom("data/220726_Miseq_micro_v2/220805_Soil_pH_adaptation_ver3/04_dada2/merged/asv_table_final.biom")
# bac_phylo=import_biom("data/220726_Miseq_micro_v2/220805_Soil_pH_adaptation_ver3/04_dada2/merged_unmerged/asv_table_final.biom")

### merge with metadata
# Import sample metadata
## in metadata erase # (This step is essential)
map <- read.table(file = 'data/220726_Miseq_micro_v2/primer_trimmed/sample_metadata_without_hashtag.tsv', sep = '\t', header = TRUE)
map <- sample_data(map)

head(map)
dim(map)
summary(map)
str(map)

summary(map)
colnames(map)
rownames(map)
nrow(map)

# Assign rownames to be Sample ID's
map$SampleID
rownames(map) <- map$SampleID
rownames(map)
dim(map)
# Merge biom data object with sample metadata + tree data(this is rooted!)
phy_tree = read_tree("data/220726_Miseq_micro_v2/220805_Soil_pH_adaptation_ver3/04_dada2/merged/rooted_tree.nwk")
phy <- merge_phyloseq(bac_phylo, map, phy_tree)

class(phy)
phy   ## 3709 otus

## changing rank names
colnames(tax_table(phy)) <- c("Kingdom", "Phylum", "Class",
                              "Order", "Family", "Genus", "Species")

sample_data(phy)
phy

lab <- c('Barneveld2_T0', 'Barneveld2_T9_H0', 'Barneveld2_T9_H0_CHL', 'Barneveld2_T9_OH10', 'Barneveld2_T9_OH10_CHL', 'LaBaghWoods1_T0', 'LaBaghWoods1_T9_H0', 'LaBaghWoods1_T9_H0_CHL', 'LaBaghWoods1_T9_OH8', 'LaBaghWoods1_T9_OH8_CHL', 'PMA_Barneveld2_T0', 'PMA_Barneveld2_T9_H0', 'PMA_Barneveld2_T9_H0_CHL', 'PMA_Barneveld2_T9_OH10', 'PMA_Barneveld2_T9_OH10_CHL', 'Negative', 'Positive')

sample_data(phy)$Sample_name <- factor(sample_data(phy)$Sample_name, levels =lab)
length(lab)

# sample_data(phy)$SampleID <- factor(sample_data(phy)$SampleID , levels =target)


phy  ## 6343 OTUs

### Usable seq analysis (18. 10. 17)
sample_names(phy)


df_phy <- psmelt(phy)

# Checking plant chloroplast or mitochondria: None 
df_phy_CP <- df_phy %>% group_by(SampleID) %>% subset(Order=='D_3__Chloroplast') %>% summarise(Chloroplast=sum(Abundance))
head(df_phy_CP, 20)

df_phy_MT <- df_phy %>% group_by(SampleID) %>% subset(Family=='D_4__Mitochondria') %>% summarise(Mitochondria=sum(Abundance))
head(df_phy_MT)

df_phy_abun <- df_phy %>% group_by(SampleID, Sample_name, Replicate) %>% summarise(Total=sum(Abundance))

# plot usable reads
df_phy_abun
df_phy_abun$Replicate <- factor(df_phy_abun$Replicate)

ggplot(df_phy_abun, aes(x=Sample_name, y=Total, fill=Replicate)) +  ## I put fill=value which was wrong
  geom_bar(stat="identity", position = "dodge") +
  # scale_fill_brewer(palette='Set2') +
  xlab("Sample") +
  ylab("Number of sequence reads \n") +
  ggtitle("Sequence Depth per sample (merged & adapter/primer rimmed)")+
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 1, vjust=0.95,angle=40,size=13,face="bold", colour = 'black'))

```

## filter negatives

```{r}

#### get unassigned vectors
#### get CP and MT phyloseq obj and vectors
# # (1) CP
# phy.cp <- subset_taxa(phy, Class == "D_2__Chloroplast") ## just confirming 
# phy.cp <- subset_taxa(phy, Order == "D_3__Chloroplast")
# vec.cp <- rownames(otu_table(phy.cp))
# length(rownames(otu_table(phy.cp))) ## 34 otus of CP
# vec.cp
# 
# # (2) MT
# # phy.mt <- subset_taxa(phy, Family == "D_4__Mitochondria")
# phy.mt <- subset_taxa(phy, Order == "D_3__Rickettsiales")
# vec.mt <- rownames(otu_table(phy.mt))
# tax_table(phy.mt)
# length(rownames(otu_table(phy.mt))) ## 78 otus of CP -> 66 after chimera filtering

# Unassigned Kingdom level
unique(tax_table(phy)[,'Kingdom']) ## only bacteria, then no need to exclude
phy.un <- subset_taxa(phy, Kingdom == "Unassigned")
vec.un <- rownames(otu_table(phy.un))
tax_table(phy.un)
length(rownames(otu_table(phy.un))) ##  479 unassigned taxa

### pop taxa application
## get rid of CP and MT otus
### pop_taxa function
pop_taxa = function(physeq, badTaxa){
  allTaxa = taxa_names(physeq)
  myTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(myTaxa, physeq))
}
### let's do it!!!

phy.clean <- pop_taxa(phy, c(vec.un))
phy  ## 3709
sum(otu_table(phy)) # 171784
phy.clean ## 3230
sum(otu_table(phy.clean)) # 170501 <- 150012

# Unassigned Phylum level
phy.phylumNA <- subset_taxa(phy.clean, is.na(Phylum))
vec.phylumNA <- rownames(otu_table(phy.phylumNA))
tax_table(phy.phylumNA)
length(rownames(otu_table(phy.phylumNA))) ##  197 phylum-unassigned taxa
sum(otu_table(phy.phylumNA))  # 945

phy.clean <- pop_taxa(phy.clean, c(vec.phylumNA))
phy  ## 3709
phy.clean ## 3033
sum(otu_table(phy.clean)) # 169556

# checking procedure of whether the MT and CP otus are cleaned
# taxa_names(subset_taxa(phy, Order == "D_3__Rickettsiales"))  # before 66
# taxa_names(subset_taxa(phy, Order=="D_3__Chloroplast")) # before 34

# taxa_names(subset_taxa(phy.clean , Order == "D_3__Rickettsiales")) # after 0
# taxa_names(subset_taxa(phy.clean , Family=="D_4__Mitochondria")) # after 0
# taxa_names(subset_taxa(phy.clean , Order == "D_3__Chloroplast")) # after 0
# phy.clean good to go sir.


#### We will also remove the "D_3__" patterns for cleaner labels
# test success
# tax_table(phy.two)[,colnames(tax_table(phy.two))] <- gsub(tax_table(phy.two)[,colnames(tax_table(phy.two))],pattern="[A-Z]_[0-9]__",replacement="")
# phy.test4 <- phy.two %>% psmelt() 
# phy.test4
tax_table(phy.clean)[,colnames(tax_table(phy.clean))] <- gsub(tax_table(phy.clean)[,colnames(tax_table(phy.clean))],pattern="[A-Z]_[0-9]__",replacement="")

#' sample_data(phy.clean)$SampleID <- factor(sample_data(phy.clean)$SampleID, levels =target_PAB)
tax_table(phy.clean)


## let's look at the negative samples
phy.clean.otu <- otu_table(phy.clean)
head(phy.clean.otu)
df.clean.otu <- data.frame(phy.clean.otu)
dim(df.clean.otu)
df.clean.otu$total <- apply(df.clean.otu, 1, sum)
head(df.clean.otu)
df.clean.otu <- tibble::rownames_to_column(df.clean.otu, var = 'OTU')

sample_names(phy.clean)
# how about we delete major taxa in the negative sequence?
negative <- df.clean.otu[,c('OTU',"SampleA53", "SampleA65")]
negative
negative$total <- apply(negative[,-1], 1, sum)
negative_0 <- subset(negative, negative$total > 0) 
neg.otu <- negative_0$OTU  ### 21 otu to be eliminated
neg.otu
length(neg.otu) # 5 strain
tax_table(phy)[neg.otu]
# library(xlsx)
# write.xlsx(tax_table(phy)[neg.otu], 'neg.otu_taxonomy.xlsx')

## checking 
## exclude all negative taxa
phy.clean.neg <- pop_taxa(phy.clean, negative_0$OTU)
phy.clean.neg  ### 3025 otu <- 3033
sum(otu_table(phy.clean.neg))  ### 163416 <- 169556

## visualize how much negative strains were present in each sample

df_phy

df_phy.clean <- psmelt(phy.clean)

df_phy.clean_neg <- df_phy.clean %>% group_by(SampleID) %>% subset(OTU %in% negative_0$OTU) %>% summarise(Negative_reads = sum(Abundance))
df_phy.clean_abun <- df_phy.clean %>% group_by(SampleID, Sample_name, Replicate, ReplicateID) %>% summarise(Total = sum(Abundance))
df_phy.clean_abun <- df_phy.clean_abun %>% left_join(df_phy.clean_neg, by = c("SampleID"="SampleID"))
df_phy.clean_abun %<>% mutate(Usable_reads = Total - Negative_reads)

l_rep <- c('Barneveld2_T0_R1', 'Barneveld2_T0_R2', 'Barneveld2_T0_R3', 'Barneveld2_T9_H0_R1', 'Barneveld2_T9_H0_R2', 'Barneveld2_T9_H0_R3', 'Barneveld2_T9_H0_CHL_R1', 'Barneveld2_T9_H0_CHL_R2', 'Barneveld2_T9_H0_CHL_R3', 'Barneveld2_T9_OH10_R1', 'Barneveld2_T9_OH10_R2', 'Barneveld2_T9_OH10_R3', 'Barneveld2_T9_OH10_CHL_R1', 'Barneveld2_T9_OH10_CHL_R2', 'Barneveld2_T9_OH10_CHL_R3', 'LaBaghWoods1_T0_R1', 'LaBaghWoods1_T0_R2', 'LaBaghWoods1_T0_R3', 'LaBaghWoods1_T9_H0_R1', 'LaBaghWoods1_T9_H0_R2', 'LaBaghWoods1_T9_H0_CHL_R1', 'LaBaghWoods1_T9_H0_CHL_R2', 'LaBaghWoods1_T9_OH8_R1', 'LaBaghWoods1_T9_OH8_R2', 'LaBaghWoods1_T9_OH8_CHL_R1', 'LaBaghWoods1_T9_OH8_CHL_R2', 'PMA_Barneveld2_T0_R1', 'PMA_Barneveld2_T0_R2', 'PMA_Barneveld2_T9_H0_R1', 'PMA_Barneveld2_T9_H0_R2', 'PMA_Barneveld2_T9_H0_CHL_R1', 'PMA_Barneveld2_T9_H0_CHL_R2', 'PMA_Barneveld2_T9_OH10_R1', 'PMA_Barneveld2_T9_OH10_R2', 'PMA_Barneveld2_T9_OH10_CHL_R1', 'PMA_Barneveld2_T9_OH10_CHL_R2', 'Negative_1', 'Negative_2', 'Positive')


colnames(df_phy.clean_abun)
df_melt_phy_clean <- melt(df_phy.clean_abun, id.vars = c("SampleID", 'Sample_name', "Replicate","ReplicateID", "Total"), variable.name = "Usable") 

df_melt_phy_clean$ReplicateID <- factor(df_melt_phy_clean$ReplicateID, levels = l_rep)

ggplot(df_melt_phy_clean, aes(x=ReplicateID, y=value, fill=Usable)) +  ## I put fill=value which was wrong
  geom_bar(stat="identity") +
  # scale_fill_brewer(palette='Set2') +
  xlab("Sample") +
  ylab("Number of sequence reads \n") +
  ggtitle("Sequence Depth per sample (After removing Kingdom/Phylum unassigned asvs)")+
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 1, vjust=0.95,angle=40,size=13,face="bold", colour = 'black'))

## remove negative samples
l_noneg <- l_rep[!(l_rep %in% c("Negative_1","Negative_2"))]
phy.clean.neg <- subset_samples(phy.clean.neg, ReplicateID %in% l_noneg)
# 37 samples

## subset samples
# phy.clean.ss <- subset_samples(phy.clean, Label %in% lb)
# (phy.clean.ss)
# phy.clean.ss <- filter_taxa(phy.clean.ss, function(x) sum(x) != 0, TRUE)
# phy.clean.ss  ## 393
# sum(otu_table(phy.clean.ss)) # 1576881

### remove otus with sequence length
## what is the taxa? over 300
# over300 = c('1866097706fa91e752c8664faf43449f', '1b2f92e822b1d78946e428e4a0cb3c0f', '3561ce5acea02d8722df1647ac04134e', '380c415e118a20e871a9212093b89dcd', '44acd89e2a099c5838d390e7fd811fb4', '7142aab19a3913224dca238c45100219', '8227de12305c6fafbcf0aeb2c7ac0a35', 'ABRL02000548.13901.15510', 'ADWL01006363.13909.15531', 'CBQP010013872.5787.7604', 'LONC01000611.65580.67188', 'MTTA01000002.173539804.173541537', 'a3aa1c5831dbdac10601b422bf713873', 'ac88961e75ff2a36681aa818ef9cc4ee', 'bb35191255f4429dad23dad82575ea78', 'd77161ae99e2715c32f077131af656ea', 'df3998cea8301598580b90cd0e0a37b6', 'e00a18e93f328393803417e3be829423', 'e74c4dd0d5f55cbba155bd31da7b7f4e', 'f248cd73405b872671c04e6bec3a530e')
# 
# tax.over300 <-  tax_table(phy)[over300]
# tax.over300 <- data.frame(tax.over300)
# is.na(tax.over300$Order)
# tax.over300$Order
# phy.clean.ss <- pop_taxa(phy.clean.ss, over300 )
# phy.clean.ss  ## 390
# sum(otu_table(phy.clean)) # 1580210

## exclude taxa less than 5 sequences
phy.clean.neg

otu_table(phy.clean.neg) %>% rowSums() %>% sort()

v.less5 <- rownames(otu_table(phy.clean.neg)[otu_table(phy.clean.neg) %>% rowSums() < 5])
v.less10 <- rownames(otu_table(phy.clean.neg)[otu_table(phy.clean.neg) %>% rowSums() < 10])
v.less20 <- rownames(otu_table(phy.clean.neg)[otu_table(phy.clean.neg) %>% rowSums() < 20])
v.less30 <- rownames(otu_table(phy.clean.neg)[otu_table(phy.clean.neg) %>% rowSums() < 30])
v.less40 <- rownames(otu_table(phy.clean.neg)[otu_table(phy.clean.neg) %>% rowSums() < 40])
v.less50 <- rownames(otu_table(phy.clean.neg)[otu_table(phy.clean.neg) %>% rowSums() < 50])

length(v.less5)   ## 1319
length(v.less10)  ## 1960
length(v.less20)  ## 2383
length(v.less30)  ## 2558
length(v.less40)  ## 2643
length(v.less50)  ## 2701

## execute
phy  ##  3709
phy.clean.neg ## 3025 negative, unassigned taxa (kingdom, phylum) removed.
(phy.clean.neg.5 <- pop_taxa(phy.clean.neg, v.less5))   # 1706 
(phy.clean.neg.10 <- pop_taxa(phy.clean.neg, v.less10)) # 1065 
(phy.clean.neg.20 <- pop_taxa(phy.clean.neg, v.less20)) # 642 
(phy.clean.neg.30 <- pop_taxa(phy.clean.neg, v.less30)) # 467 
(phy.clean.neg.40 <- pop_taxa(phy.clean.neg, v.less40)) # 382 
(phy.clean.neg.50 <- pop_taxa(phy.clean.neg, v.less50)) # 324 

sum(otu_table(phy)) ## 171784 otu reads.
sum(otu_table(phy.clean)) ## 169556
sum(otu_table(phy.clean.neg)) ## 163416
sum(otu_table(phy.clean.neg.5)) ## 159746
sum(otu_table(phy.clean.neg.10)) ## 155540
sum(otu_table(phy.clean.neg.20)) ## 149716

## length analysis
tsv_mergetrimmed <- read_tsv("data/220726_Miseq_micro_v2/asv_length_merged_primer_trimmed.tsv", col_names = F) %>% rename(ASV = X1, Length = X2)

df_phy_mergetrimmed <- psmelt(phy.clean.neg)
df_mergetrimmed <- df_phy_mergetrimmed %>% select(OTU, Sample, Abundance, Replicate) %>% group_by(OTU) %>% summarise(Frequency=sum(Abundance)) %>% rename(ASV = OTU) %>% left_join(tsv_mergetrimmed, by=c("ASV"="ASV")) %>% ungroup()
df_mergetrimmed_len <- df_mergetrimmed %>% group_by(Length) %>% summarize(Length_frequency = sum(Frequency)) %>% ungroup()

df_mergetrimmed_len %<>% mutate(Ranges = cut(Length, seq(0, 500, 20))) %>% group_by(Ranges) %>% 
  summarize(sums = sum(Length_frequency))

ggplot(df_mergetrimmed_len, aes(x=Ranges, y=sums)) +
  geom_bar(stat="identity", fill = "red", alpha = 0.5) +
  xlab("\n Asv sequence length") +
  ylab("Total read count (all samples combined) \n") +
  ggtitle("Read length distribution (After merged & adapter/primer trimmed)") +
  mytheme_2d


```

## plot relative abundance plot

```{r}
## average or sum replicates
# phy.m <-  merge_samples(phy.clean.neg, "Sample_name")
# tax_table(phy.m)

# we are going to plot replicates
l_rep <- c('Barneveld2_T0_R1', 'Barneveld2_T0_R2', 'Barneveld2_T0_R3', 'Barneveld2_T9_H0_R1', 'Barneveld2_T9_H0_R2', 'Barneveld2_T9_H0_R3', 'Barneveld2_T9_H0_CHL_R1', 'Barneveld2_T9_H0_CHL_R2', 'Barneveld2_T9_H0_CHL_R3', 'Barneveld2_T9_OH10_R1', 'Barneveld2_T9_OH10_R2', 'Barneveld2_T9_OH10_R3', 'Barneveld2_T9_OH10_CHL_R1', 'Barneveld2_T9_OH10_CHL_R2', 'Barneveld2_T9_OH10_CHL_R3', 'LaBaghWoods1_T0_R1', 'LaBaghWoods1_T0_R2', 'LaBaghWoods1_T0_R3', 'LaBaghWoods1_T9_H0_R1', 'LaBaghWoods1_T9_H0_R2', 'LaBaghWoods1_T9_H0_CHL_R1', 'LaBaghWoods1_T9_H0_CHL_R2', 'LaBaghWoods1_T9_OH8_R1', 'LaBaghWoods1_T9_OH8_R2', 'LaBaghWoods1_T9_OH8_CHL_R1', 'LaBaghWoods1_T9_OH8_CHL_R2', 'PMA_Barneveld2_T0_R1', 'PMA_Barneveld2_T0_R2', 'PMA_Barneveld2_T9_H0_R1', 'PMA_Barneveld2_T9_H0_R2', 'PMA_Barneveld2_T9_H0_CHL_R1', 'PMA_Barneveld2_T9_H0_CHL_R2', 'PMA_Barneveld2_T9_OH10_R1', 'PMA_Barneveld2_T9_OH10_R2', 'PMA_Barneveld2_T9_OH10_CHL_R1', 'PMA_Barneveld2_T9_OH10_CHL_R2', 'Positive')

## let's get phylum sir
df.phylum <- phy.clean.neg %>%
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  psmelt()
df.phylum$ReplicateID <- factor(df.phylum$ReplicateID, levels = l_rep)

library(forcats) 
df.phylum %<>% mutate(Phylum = fct_explicit_na(Phylum, na_level = "unidentified"))
unique(df.phylum$Phylum)

levels(df.phylum$Phylum)
levels(df.phylum$Phylum) = c(levels(df.phylum$Phylum), 'Low abundance')

# we need to group by samples
df.phylum.rel <- df.phylum %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

# df.phylum.rel[df.phylum.rel$RelAbundance < 5,]$Phylum <- 'Low abundance'

### let's test
ord <- df.phylum.rel %>% group_by(Phylum) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Phylum
rev(tail(vec,10))
vec
# x = factor(x,levels(x)[c(4,5,1:3)])
# levels(vec) <- c(levels(vec), 'Low abundance')
# vec
# length(vec)
# levels(vec) <- c(length(vec),1:length(vec)-1)
# vec

vec
# rest <- vec[!vec %in% c('Low abundance')]
# vec <- factor(vec,levels =  )
rev(vec)
# New

# vec <- c('Low abundance','unidentified','Acidobacteria','Bacteroidetes','Firmicutes','Actinobacteria','Proteobacteria')


df.phylum.rel$Phylum <- factor(df.phylum.rel$Phylum, levels = vec)

## relative abundance with less than 0.5% in sample was labeled 'low abundance'

## plot relative abundance
ggplot(df.phylum.rel, aes(x=ReplicateID, y = RelAbundance, fill = Phylum)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection) +
  
  xlab('')+
  ylab("Relative Abundance (%) \n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(nrow = 4,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())


##### now class
## let's get class sir
df.class <- phy.clean.neg %>%
  tax_glom(taxrank = "Class", NArm = FALSE) %>% 
  psmelt()
df.class$ReplicateID <- factor(df.class$ReplicateID, levels = l_rep)

library(forcats) 
df.class %<>% mutate(Class = fct_explicit_na(Class, na_level = "unidentified"))
unique(df.class$Class)

levels(df.class$Class)
levels(df.class$Class) = c(levels(df.class$Class), 'Low abundance')

# we need to group by samples
df.class.rel <- df.class %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.class.rel[df.class.rel$RelAbundance < 1,]$Class <- 'Low abundance'

### let's test
ord <- df.class.rel %>% group_by(Class) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Class
# rev(tail(vec,10))
# vec
# write.table(vec,'vec.txt', quote=FALSE, sep='\t', row.names = F, col.names = F)

df.class.rel$Class <- factor(df.class.rel$Class, levels = vec)

## 2. plot relative abundance
ggplot(df.class.rel, aes(x=ReplicateID, y = RelAbundance, fill = Class)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(nrow = 3,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())

##### now Order

df.order <- phy.clean.neg %>%
  tax_glom(taxrank = "Order", NArm = FALSE) %>% 
  psmelt()
df.order$ReplicateID <- factor(df.order$ReplicateID, levels = l_rep)

library(forcats) 
df.order %<>% mutate(Order = fct_explicit_na(Order, na_level = "unidentified"))
unique(df.order$Order)

levels(df.order$Order) = c(levels(df.order$Order), 'Low abundance')

# we need to group by samples
df.order.rel <- df.order %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.order.rel[df.order.rel$RelAbundance < 0.3,]$Order <- 'Low abundance'

### let's test
ord <- df.order.rel %>% group_by(Order) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Order
# rev(tail(vec,10))
# vec
# write.table(vec,'vec.txt', quote=FALSE, sep='\t', row.names = F, col.names = F)

df.order.rel$Order <- factor(df.order.rel$Order, levels = vec)
length(vec)
## 2. plot relative abundance
ggplot(df.order.rel, aes(x=ReplicateID, y = RelAbundance, fill = Order)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(ncol = 6,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())


##### now family
df.family <- phy.clean.neg %>%
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  psmelt()
df.family$ReplicateID <- factor(df.family$ReplicateID, levels = l_rep)

library(forcats) 
df.family %<>% mutate(Family = fct_explicit_na(Family, na_level = "unidentified"))
unique(df.family$Family)

levels(df.family$Family) = c(levels(df.family$Family), 'Low abundance')

# we need to group by samples
df.family.rel <- df.family %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.family.rel[df.family.rel$RelAbundance < 0.3,]$Family <- 'Low abundance'


### let's test
ord <- df.family.rel %>% group_by(Family) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Family
length(vec)
# rev(tail(vec,10))
# vec
# length(vec)
# write.table(vec,'vec.txt', quote=FALSE, sep='\t', row.names = F, col.names = F)
df.family.rel$Family <- factor(df.family.rel$Family, levels = vec)

## 2. plot relative abundance
ggplot(df.family.rel, aes(x=ReplicateID, y = RelAbundance, fill = Family)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection2) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(ncol = 7,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())



##### now Genus
### let's get phylum sir
# prune out phyla below 2% in each sample
df.genus <- phy.clean.neg %>%
  tax_glom(taxrank = "Genus", NArm = FALSE) %>% 
  psmelt()
df.genus$ReplicateID <- factor(df.genus$ReplicateID, levels = l_rep)

# library(dplyr)
# library(forcats) 
# library(magrittr)
df.genus %<>% mutate(Genus = fct_explicit_na(Genus, na_level = "unidentified"))
unique(df.genus$Genus)

levels(df.genus$Genus) = c(levels(df.genus$Genus), 'Low abundance')

# we need to group by samples
df.genus.rel <- df.genus %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.genus.rel[df.genus.rel$RelAbundance < 0.3,]$Genus <- 'Low abundance'

### let's test

gen.ord <- df.genus.rel %>% group_by(Genus) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- gen.ord$Genus
length(vec)
# rev(tail(vec.gen,10))
# vec.gen
# write.xlsx(rev(vec.gen), 'bacteria_genus_list.xlsx')

# let's make list of genus starting from kingdom
df.genus.rel$Genus <- factor(df.genus.rel$Genus, levels = vec)

genus_color <- c(
  'gray','black',"#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  
  '#7ac864', '#ff7473', '#00e2ff', '#8836ff', '#732b90',
  '#e72fff', '#006eff', '#e7eb1c','#00ffe0','#ffc680',
  '#ABD0BC','#519eff', '#ffaa00','#00a300','#b23b00',
  '#ecb0e7','#ff5a00' ,'#8ee7e7'
)
## 2. plot relative abundance
ggplot(df.genus.rel, aes(x=ReplicateID, y = RelAbundance, fill = Genus)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection2) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(ncol = 7,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())


# plot only the positive
df.genus.rel.positive = df.genus.rel %>% filter(ReplicateID == "Positive")

ggplot(df.genus.rel.positive, aes(x=ReplicateID, y = RelAbundance, fill = Genus)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection2) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  guides(fill = guide_legend(ncol = 2,reverse = T)) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  mytheme_2d +
  theme(legend.position="bottom")

```


## Plot ASV rank abundance

```{r}
## convert OTU table into relative abundance
# how about using phy.clean.neg.10
otu_rel <- otu_table(phy.clean.neg.10) # remove asv that has less than 10 reads total.
class(otu_rel)

hist(rowSums(otu_rel), breaks = 60)
table(rowSums(otu_rel))
sort(rowSums(otu_rel)) %>% tail()

mat_rel <- apply(otu_rel,2, function(x) x/sum(x))
dim(mat_rel) # 1903 asv
colSums(mat_rel)

## Reorder the rows and change rownames
sort(rowSums(otu_rel), decreasing = T)
names(sort(rowSums(otu_rel), decreasing = T))

mat_rel2 <- mat_rel[names(sort(rowSums(otu_rel), decreasing = T)),,drop=F]
rownames(mat_rel2) <- paste0("asv", sprintf("%04d", 1:dim(mat_rel2)[1]))

mat_rel2
colSums(mat_rel2)
dim(mat_rel2)

## rank matrix per column
library(matrixStats)

## how to calculate ranks
x <- cbind(a=c(11,22,44),
       b=c(21,12,NA),
       c=c(35,66,12))
x
# apply(x, 2, rank)
apply(-x, 2, rank)

# make 0 NA
any(is.na(mat_rel2)) # no NA in the data
mat_rel2[mat_rel2 == 0] <- NA
dim(mat_rel2)
sum(is.na(mat_rel2)) # 67553 NAs

mat_rank <- apply(-mat_rel2, 2, rank)
any(is.na(mat_rank)) # there should be NA
dim(mat_rank)
# change back to NA, which are supposed to be NA
mat_rank[is.na(mat_rel2)] <- NA
any(is.na(mat_rank))
sum(is.na(mat_rank)) # 34987 NAs

colMaxs(mat_rank, na.rm =T)
colMins(mat_rank, na.rm =T)

mat_rank[,1]
table(mat_rank[,1])
table(mat_rank[,2])
table(mat_rank[,3])
table(mat_rank[,4]) %>% tail()

mat_rank

## plot rank
sub_mat_rank <- mat_rank[,1:2]
mlt_rank <- melt(sub_mat_rank)
colnames(mlt_rank) <- c("ASV", "Sample", "Rank")
head(mlt_rank)
class(mlt_rank)

## get the rank abundance aligned to SampleA1
vec_asv_sampleA1 <- mlt_rank %>% filter(Sample == "SampleA1") %>% arrange(Rank) %>% select(ASV)
head(vec_asv_sampleA1)
vec_asv_sampleA1 <- vec_asv_sampleA1[,1]
mlt_rank$ASV <- factor(mlt_rank$ASV, levels = vec_asv_sampleA1)

ggplot(mlt_rank, aes(x = ASV, y = Rank)) +
  geom_line(aes(col=Sample)) +
  geom_point(aes(col=Sample)) +
  ylab("Rank \n") +
  xlab(paste0("\n ASV ID")) +
  mytheme_2d +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
  # scale_y_continuous(expand = c(0, 0))

## Okay this is a good demonstration of the data


```

## Let's begin the proper analysis

```{r}
## connect metadata to the asv table

phy.clean.neg.10
df_meta <- sample_data(phy.clean.neg.10)

## melt the total rank matrix into dataframe, then attach the metadata
mat_rank
mlt_rank <- melt(mat_rank)
colnames(mlt_rank) <- c("ASV", "Sample", "Rank")
head(mlt_rank)

dim(mlt_rank) #70411 3
mlt_rank_meta <- mlt_rank %>% left_join(df_meta, by=c("Sample"="SampleID"))
dim(mlt_rank_meta) #70411 14
colnames(mlt_rank_meta)

# replicate check - start with 1 sample
mlt_rank_meta %>% select(Sample_name) %>% unique()
mlt_B2_T0 <- mlt_rank_meta %>% filter(Sample_name == "Barneveld2_T0") 
dim(mlt_B2_T0)
head(mlt_B2_T0)


# plot
## get the rank abundance aligned to Barneveld2_T0
vec_asv_B2T0 <- mlt_B2_T0 %>% filter(Replicate == 1) %>% arrange(Rank) %>% select(ASV)
dim(vec_asv_B2T0)
head(vec_asv_B2T0)
vec_asv_B2T0 <- vec_asv_B2T0[,1]
mlt_B2_T0$ASV <- factor(mlt_B2_T0$ASV, levels = vec_asv_B2T0)

mlt_B2_T0$Replicate <- factor(mlt_B2_T0$Replicate)　　　

ggplot(mlt_B2_T0 %>% filter(Replicate ==1), aes(x = ASV, y = Rank)) +
  # geom_line(aes(col=Sample)) +
  geom_point(aes(col=Replicate)) +
  ggtitle("Barneveld2_T0") +
  ylab("Rank \n") +
  xlab("\n ASV ID") +
  mytheme_2d +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())

# what is happening here?
ggplot(mlt_B2_T0, aes(x = ASV, y = Rank)) +
  # geom_line(aes(col=Sample)) +
  geom_point(aes(col=Replicate)) +
  ggtitle("Barneveld2_T0") +
  ylab("Rank \n") +
  xlab("\n ASV ID") +
  mytheme_2d +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
  # scale_y_continuous(expand = c(0, 0))

# upto rank 500
# 
# mlt_B2_T0_rank500 <- mlt_B2_T0 %>% filter(Rank < 500)
# vec_B2_T0_rank500 <- mlt_B2_T0_rank500 %>% filter(Replicate == 1) %>% arrange(Rank) %>% select(ASV)
# head(vec_B2_T0_rank500)
# vec_B2_T0_rank500 <- vec_B2_T0_rank500[,1]
# mlt_B2_T0_rank500$ASV <- factor(mlt_B2_T0_rank500$ASV, levels = vec_B2_T0_rank500)
# 
# 
# ggplot(mlt_B2_T0_rank500, aes(x = ASV, y = Rank)) +
#   # geom_line(aes(col=Replicate)) +
#   geom_point(aes(col=Replicate)) +
#   ylab("Rank \n") +
#   xlab("\n ASV ID") +
#   ggtitle("Barneveld2_T0 zoom-in rank500") +
#   mytheme_2d +
#   theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())

## plot across different samples, align asv to replicate 1
l_sample_names <- mlt_rank_meta %>% select(Sample_name) %>% unique()
l_sample_names <- l_sample_names[,1]
l_sample_names <- l_sample_names[-length(l_sample_names)] # remove positive sample

plot_rank_asv <- function(mlt_rank_meta, Sample_name_i = "Barneveld2_T0", Replicate_i = 1){
  mlt_B2_T0 <- mlt_rank_meta %>% filter(Sample_name == Sample_name_i) 
  dim(mlt_B2_T0)
  head(mlt_B2_T0)
  
  ## get the rank abundance aligned to Barneveld2_T0
  vec_asv_B2T0 <- mlt_B2_T0 %>% filter(Replicate == Replicate_i) %>% arrange(Rank) %>% select(ASV)
  dim(vec_asv_B2T0)
  head(vec_asv_B2T0)
  vec_asv_B2T0 <- vec_asv_B2T0[,1]
  mlt_B2_T0$ASV <- factor(mlt_B2_T0$ASV, levels = vec_asv_B2T0)
  mlt_B2_T0$Replicate <- factor(mlt_B2_T0$Replicate)　　　
  
  ggplot(mlt_B2_T0, aes(x = ASV, y = Rank)) +
    # geom_line(aes(col=Sample)) +
    geom_point(aes(col=Replicate)) +
    ggtitle(paste0(Sample_name_i)) +
    ylab("Rank") +
    xlab("ASV ID") +
    mytheme_2d +
    theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
    # scale_y_continuous(expand = c(0, 0))
}

# plot in 1 sheet
l_plot <- list()
for (i in 1:length(l_sample_names)) {
    print(i)
    l_plot[[i]] <- plot_rank_asv(mlt_rank_meta, l_sample_names[i], Replicate_i=1)
}

library(gridExtra)
grid.arrange(grobs = l_plot, ncol = 5)

l_plot <- list()
for (i in 1:length(l_sample_names)) {
    print(i)
    l_plot[[i]] <- plot_rank_asv(mlt_rank_meta, l_sample_names[i], Replicate_i=2)
}

library(gridExtra)
grid.arrange(grobs = l_plot, ncol = 5)

l_plot <- list()
for (i in 1:length(l_sample_names)) {
    print(i)
    l_plot[[i]] <- plot_rank_asv(mlt_rank_meta, l_sample_names[i], Replicate_i=3)
}

library(gridExtra)
grid.arrange(grobs = l_plot, ncol = 5)
## plot abundance - rank plot to see how much abundance is rank 500
mat_rel2
mlt_rank_meta
colnames(mlt_rank_meta)

mlt_rel2 <- melt(mat_rel2)
colnames(mlt_rel2) <- c("ASV", "Sample", "Rel_abundance")
head(mlt_rel2)

# join two dataframes
mlt_rank_rel <- mlt_rank_meta %>% left_join(mlt_rel2, by = c("ASV"="ASV", "Sample"="Sample"))
colnames(mlt_rank_rel)
# plot Rank (x) - relative abundance (y) plots
l_sample_names

mlt_rank_rel_B2_T0 <- mlt_rank_rel %>% filter(Sample_name == "Barneveld2_T0")
mlt_rank_rel_B2_T0$Replicate <- factor(mlt_rank_rel_B2_T0$Replicate)

dim(mlt_rank_rel_B2_T0)

# what is happening here?
mlt_rank_rel_B2_T0 %>% select(Rank, Rel_abundance) # oh no... got ranked as well...


dim(na.omit(mlt_rank_rel_B2_T0))
mlt_rank_rel_B2_T0_noNA <- na.omit(mlt_rank_rel_B2_T0)

ggplot(mlt_rank_rel_B2_T0_noNA, aes(x = Rank, y = Rel_abundance)) +
  geom_line(aes(col=Replicate)) +
  geom_point(aes(col=Replicate)) +
  ggtitle(paste0("Barneveld2_T0")) +
  ylab("Relative abundance") +
  xlab("Rank") +
  mytheme_2d

mlt_rank_rel_B2_T0_noNA %>% group_by(Replicate) %>% summarize(Total = sum(Rel_abundance))

## plot Rel_abundance - rank plots for all samples
plot_rank_abundance <- function(mlt_rank_rel, Sample_name_i){
  mlt_rank_rel_B2_T0 <- mlt_rank_rel %>% filter(Sample_name == Sample_name_i)
  
  mlt_rank_rel_B2_T0$Replicate <- factor(mlt_rank_rel_B2_T0$Replicate)
  
  mlt_rank_rel_B2_T0_noNA <- na.omit(mlt_rank_rel_B2_T0)
  df_mlt_plot <- mlt_rank_rel_B2_T0_noNA %>% select(Sample_name, Replicate, Rank, Rel_abundance) %>% ungroup()
  colnames(df_mlt_plot)
  class(df_mlt_plot)
  # print("we are good")
  
  ggplot(df_mlt_plot, aes(x=Rank, y=Rel_abundance)) +
    geom_point(aes(col=Replicate)) +
    geom_line(aes(col=Replicate)) +
    ggtitle(paste0(Sample_name_i)) +
    ylab("Relative abundance") +
    xlab("Rank") +
    mytheme_2d
}

# plot in 1 sheet
l_plot2 <- list()
for (i in 1:length(l_sample_names)) {
  print(i)
  # print(l_sample_names[i])
  l_plot2[[i]] <- plot_rank_abundance(mlt_rank_rel, l_sample_names[i])
}

library(gridExtra)
grid.arrange(grobs = l_plot2, ncol = 5)

```


## Let's visualize the overlap between replicates

```{r}
phy.clean.neg.10

df_subset <- psmelt(phy.clean.neg.10) %>%  select(OTU, Sample, Replicate, Abundance) %>% group_by(Sample) %>%  filter(Abundance > 0) %>% arrange(Sample)

dim(df_subset)

df_otu_1 <- df_subset %>% filter(Replicate == 1) %>% select(OTU) %>% unique()
df_otu_1$OTU

venn_each_sample <- function(phy.clean.neg.10){
  library(VennDiagram)
  df_subset <- psmelt(phy.clean.neg.10) %>%  select(OTU, Sample, Replicate, Abundance) %>% group_by(Sample) %>%  filter(Abundance > 0) %>% arrange(Sample)
  df_otu_1 <- df_subset %>% filter(Replicate == 1) %>% select(OTU) %>% unique()
  vec.f <- df_otu_1$OTU
  df_otu_2 <- df_subset %>% filter(Replicate == 2) %>% select(OTU) %>% unique()
  vec.s <- df_otu_2$OTU
  df_otu_3 <- df_subset %>% filter(Replicate == 3) %>% select(OTU) %>% unique()
  vec.t <- df_otu_3$OTU
  n12 <- intersect(vec.f,vec.s)
  n23 <- intersect(vec.s,vec.t)
  n13 <- intersect(vec.f,vec.t)
  n123 <- intersect(intersect(vec.f,vec.s),vec.t)
  
  my.list <- list(first = vec.f,second = vec.s, third=vec.t, n12=n12, n23=n23,n13=n13,n123=n123)
  my.list
  draw.triple.venn(area1 = length(my.list$first),
                   area2 = length(my.list$second),
                   area3 = length(my.list$third),
                   n12 = length(my.list$n12),
                   n23 = length(my.list$n23),
                   n13 = length(my.list$n13),
                   n123 = length(my.list$n123),
                   #category = c("first","second",'third'),
                   fill = c("#FE9696",'#96FE96',"#9696FE"),
                   euler.d = F,
                   scaled = F,
                   ext.text = T,
                   ext.percent = c(0.1,0.1,0.1),
                   ext.length = 1,
                   label.col = rep("gray10",7),
                   lwd = 0,
                   cex = 1,
                   fontface = rep("plain",7),
                   fontfamily = rep("sans",7), 
                   # cat.cex = 1.5,
                   # cat.fontface = rep("bold",3),
                   # cat.fontfamily = rep("sans",3),
                   # #cat.pos = c(0, 0,0),
                   print.mode = c("percent","raw")
  )
}





l_venn <- list()
for (i in 1:length(l_sample_names)){
  print(i)
  j <- l_sample_names[i]
  phy.ss1 <- subset_samples(phy.clean.neg.10, Sample_name == j)
  l_venn[[i]] <- grid.arrange(gTree(children=venn_each_sample(phy.ss1)), top = paste0(j))
}
length(l_venn)

grid.arrange(grobs = l_venn, ncol=5)

```


## Let's do this OTU level

OTU level

```{r}
### We can then load the biom file with phyloseq function import_biom. We extract the OTU table with OTU abundances and the taxonomy table from the resulting phyloseq object.
bac_phylo=import_biom("data/220726_Miseq_micro_v2/220805_Soil_pH_adaptation_ver3/06_Final_OTU_chimera_filtered/merged/OTU_table_final.biom")

### merge with metadata
# Import sample metadata
## maybe Gyeryueng data didn't work because it wasn't transformed to json file

## in metadata erase # (This step is essential)
map <- read.table(file = 'data/220726_Miseq_micro_v2/merged_OTU/sample_metadata_without_hashtag.tsv', sep = '\t', header = TRUE)
map <- sample_data(map)
head(map)
dim(map)
summary(map)
str(map)

summary(map)
colnames(map)
rownames(map)
nrow(map)

# Assign rownames to be Sample ID's
map$SampleID
rownames(map) <- map$SampleID
rownames(map)
dim(map)
# Merge biom data object with sample metadata + tree data(this is rooted!)
phy_tree = read_tree("data/220726_Miseq_micro_v2/220805_Soil_pH_adaptation_ver3/06_Final_OTU_chimera_filtered/merged/rooted_tree.nwk")
phy_otu <- merge_phyloseq(bac_phylo, map, phy_tree) # merged OTU

class(phy_otu)
phy_otu   ## 2422 otus

## changing rank names
na.omit(tax_table(phy_otu))

## No spiecies for OTU?
colnames(tax_table(phy_otu)) <- c("Kingdom", "Phylum", "Class",
                              "Order", "Family", "Genus", "Species")

sample_data(phy_otu)
phy_otu

lab <- c('Barneveld2_T0', 'Barneveld2_T9_H0', 'Barneveld2_T9_H0_CHL', 'Barneveld2_T9_OH10', 'Barneveld2_T9_OH10_CHL', 'LaBaghWoods1_T0', 'LaBaghWoods1_T9_H0', 'LaBaghWoods1_T9_H0_CHL', 'LaBaghWoods1_T9_OH8', 'LaBaghWoods1_T9_OH8_CHL', 'PMA_Barneveld2_T0', 'PMA_Barneveld2_T9_H0', 'PMA_Barneveld2_T9_H0_CHL', 'PMA_Barneveld2_T9_OH10', 'PMA_Barneveld2_T9_OH10_CHL', 'Negative', 'Positive')

sample_data(phy_otu)$Sample_name <- factor(sample_data(phy_otu)$Sample_name, levels =lab)
length(lab)

# sample_data(phy)$SampleID <- factor(sample_data(phy)$SampleID , levels =target)


phy_otu  ## 2422 OTUs

### Usable seq analysis (18. 10. 17)
sample_names(phy_otu)

df_phy <- psmelt(phy_otu)

# Checking plant chloroplast or mitochondria: None 
df_phy_CP <- df_phy %>% group_by(SampleID) %>% subset(Order=='D_3__Chloroplast') %>% summarise(Chloroplast=sum(Abundance))
head(df_phy_CP, 20)

df_phy_MT <- df_phy %>% group_by(SampleID) %>% subset(Family=='D_4__Mitochondria') %>% summarise(Mitochondria=sum(Abundance))
head(df_phy_MT)

df_phy_abun <- df_phy %>% group_by(SampleID, Sample_name, Replicate) %>% summarise(Total=sum(Abundance))

# plot usable reads
df_phy_abun
df_phy_abun$Replicate <- factor(df_phy_abun$Replicate)

ggplot(df_phy_abun, aes(x=Sample_name, y=Total, fill=Replicate)) +  ## I put fill=value which was wrong
  geom_bar(stat="identity", position = "dodge") +
  # scale_fill_brewer(palette='Set2') +
  xlab("Sample") +
  ylab("Number of sequence reads \n") +
  ggtitle("Sequence Depth per sample (After removing Kingdom/Phylum unassigned OTUs)")+
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 1, vjust=0.95,angle=40,size=13,face="bold", colour = 'black'))

```

## filter negatives

```{r}

#### get unassigned vectors
#### get CP and MT phyloseq obj and vectors
# # (1) CP
# phy.cp <- subset_taxa(phy, Class == "D_2__Chloroplast") ## just confirming 
# phy.cp <- subset_taxa(phy, Order == "D_3__Chloroplast")
# vec.cp <- rownames(otu_table(phy.cp))
# length(rownames(otu_table(phy.cp))) ## 34 otus of CP
# vec.cp
# 
# # (2) MT
# # phy.mt <- subset_taxa(phy, Family == "D_4__Mitochondria")
# phy.mt <- subset_taxa(phy, Order == "D_3__Rickettsiales")
# vec.mt <- rownames(otu_table(phy.mt))
# tax_table(phy.mt)
# length(rownames(otu_table(phy.mt))) ## 78 otus of CP -> 66 after chimera filtering

# Unassigned Kingdom level
unique(tax_table(phy_otu)[,'Kingdom']) ## only bacteria, then no need to exclude
phy.un <- subset_taxa(phy_otu, Kingdom == "Unassigned")
vec.un <- rownames(otu_table(phy.un))
tax_table(phy.un)
length(rownames(otu_table(phy.un)))  ##  455 unassigned taxa
rownames(otu_table(phy.un)) %>% unique()
sum(otu_table(phy.un)) # 1281 reads

### pop taxa application
## get rid of CP and MT otus
### pop_taxa function
pop_taxa = function(physeq, badTaxa){
  allTaxa = taxa_names(physeq)
  myTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
  return(prune_taxa(myTaxa, physeq))
}
### let's do it!!!

phy_otu.clean <- pop_taxa(phy_otu, c(vec.un))
phy_otu  ## 2422
sum(otu_table(phy_otu)) # 170857

phy_otu.clean ## 1967
sum(otu_table(phy_otu.clean)) # 169576

# Unassigned Phylum level
phy_otu.phylumNA <- subset_taxa(phy_otu.clean, is.na(Phylum))
vec.phylumNA <- rownames(otu_table(phy_otu.phylumNA))
tax_table(phy_otu.phylumNA)
length(rownames(otu_table(phy_otu.phylumNA))) ##  159 unassigned taxa
sum(otu_table(phy_otu.phylumNA)) # 880

phy_otu.clean <- pop_taxa(phy_otu.clean, c(vec.phylumNA))
phy_otu  ## 2422
phy_otu.clean ## 1808
sum(otu_table(phy_otu.clean)) # 168696

# checking procedure of whether the MT and CP otus are cleaned
# taxa_names(subset_taxa(phy, Order == "D_3__Rickettsiales"))  # before 66
# taxa_names(subset_taxa(phy, Order=="D_3__Chloroplast")) # before 34

# taxa_names(subset_taxa(phy.clean , Order == "D_3__Rickettsiales")) # after 0
# taxa_names(subset_taxa(phy.clean , Family=="D_4__Mitochondria")) # after 0
# taxa_names(subset_taxa(phy.clean , Order == "D_3__Chloroplast")) # after 0
# phy.clean good to go sir.


#### We will also remove the "D_3__" patterns for cleaner labels
# test success
# tax_table(phy.two)[,colnames(tax_table(phy.two))] <- gsub(tax_table(phy.two)[,colnames(tax_table(phy.two))],pattern="[A-Z]_[0-9]__",replacement="")
# phy.test4 <- phy.two %>% psmelt() 
# phy.test4
tax_table(phy_otu.clean)[,colnames(tax_table(phy_otu.clean))] <- gsub(tax_table(phy_otu.clean)[,colnames(tax_table(phy_otu.clean))],pattern="[A-Z]_[0-9]__",replacement="")

#' sample_data(phy.clean)$SampleID <- factor(sample_data(phy.clean)$SampleID, levels =target_PAB)
tax_table(phy_otu.clean)


## let's look at the negative samples
phy_otu.clean.otu <- otu_table(phy_otu.clean)
head(phy_otu.clean.otu)
df.clean.otu <- data.frame(phy_otu.clean.otu)
dim(df.clean.otu)
df.clean.otu$total <- apply(df.clean.otu, 1, sum)
head(df.clean.otu)
df.clean.otu <- tibble::rownames_to_column(df.clean.otu, var = 'OTU')

sample_names(phy_otu.clean)
# how about we delete major taxa in the negative sequence?
negative <- df.clean.otu[,c('OTU',"SampleA53", "SampleA65")]
negative
negative$total <- apply(negative[,-1], 1, sum)
negative_0 <- subset(negative, negative$total > 0) 
neg.otu <- negative_0$OTU  ### 21 otu to be eliminated
neg.otu
length(neg.otu) # 6 strain
tax_table(phy_otu)[neg.otu]
# library(xlsx)
# write.xlsx(tax_table(phy)[neg.otu], 'neg.otu_taxonomy.xlsx')

## checking 
## exclude all negative taxa
phy_otu.clean.neg <- pop_taxa(phy_otu.clean, negative_0$OTU)
phy_otu.clean.neg  ### 1802 otu <- 1808
sum(otu_table(phy_otu.clean.neg))  ### 158624 <- 168696

## visualize how much negative strains were present in each sample

df_phy_otu.clean <- psmelt(phy_otu.clean)

df_phy_otu.clean_neg <- df_phy_otu.clean %>% group_by(SampleID) %>% subset(OTU %in% negative_0$OTU) %>% summarise(Negative_reads = sum(Abundance))
df_phy_otu.clean_abun <- df_phy_otu.clean %>% group_by(SampleID, Sample_name, Replicate, ReplicateID) %>% summarise(Total = sum(Abundance))
df_phy_otu.clean_abun <- df_phy_otu.clean_abun %>% left_join(df_phy_otu.clean_neg, by = c("SampleID"="SampleID"))
df_phy_otu.clean_abun %<>% mutate(Usable_reads = Total - Negative_reads)

l_rep <- c('Barneveld2_T0_R1', 'Barneveld2_T0_R2', 'Barneveld2_T0_R3', 'Barneveld2_T9_H0_R1', 'Barneveld2_T9_H0_R2', 'Barneveld2_T9_H0_R3', 'Barneveld2_T9_H0_CHL_R1', 'Barneveld2_T9_H0_CHL_R2', 'Barneveld2_T9_H0_CHL_R3', 'Barneveld2_T9_OH10_R1', 'Barneveld2_T9_OH10_R2', 'Barneveld2_T9_OH10_R3', 'Barneveld2_T9_OH10_CHL_R1', 'Barneveld2_T9_OH10_CHL_R2', 'Barneveld2_T9_OH10_CHL_R3', 'LaBaghWoods1_T0_R1', 'LaBaghWoods1_T0_R2', 'LaBaghWoods1_T0_R3', 'LaBaghWoods1_T9_H0_R1', 'LaBaghWoods1_T9_H0_R2', 'LaBaghWoods1_T9_H0_CHL_R1', 'LaBaghWoods1_T9_H0_CHL_R2', 'LaBaghWoods1_T9_OH8_R1', 'LaBaghWoods1_T9_OH8_R2', 'LaBaghWoods1_T9_OH8_CHL_R1', 'LaBaghWoods1_T9_OH8_CHL_R2', 'PMA_Barneveld2_T0_R1', 'PMA_Barneveld2_T0_R2', 'PMA_Barneveld2_T9_H0_R1', 'PMA_Barneveld2_T9_H0_R2', 'PMA_Barneveld2_T9_H0_CHL_R1', 'PMA_Barneveld2_T9_H0_CHL_R2', 'PMA_Barneveld2_T9_OH10_R1', 'PMA_Barneveld2_T9_OH10_R2', 'PMA_Barneveld2_T9_OH10_CHL_R1', 'PMA_Barneveld2_T9_OH10_CHL_R2', 'Negative_1', 'Negative_2', 'Positive')


colnames(df_phy_otu.clean_abun)

df_melt_phy_otu_clean <- melt(df_phy_otu.clean_abun, id.vars = c("SampleID", 'Sample_name', "Replicate","ReplicateID", "Total"), variable.name = "Usable") 

df_melt_phy_otu_clean$ReplicateID <- factor(df_melt_phy_otu_clean$ReplicateID, levels = l_rep)

ggplot(df_melt_phy_otu_clean, aes(x=ReplicateID, y=value, fill=Usable)) +  ## I put fill=value which was wrong
  geom_bar(stat="identity") +
  # scale_fill_brewer(palette='Set2') +
  xlab("Sample") +
  ylab("Number of sequence reads \n") +
  ggtitle("Sequence Depth per sample (After removing Kingdom/Phylum unassigned OTUs)")+
  mytheme_2d +
  theme(axis.text.x = element_text(hjust = 1, vjust=0.95,angle=40,size=13,face="bold", colour = 'black'))

## remove negative samples
l_noneg <- l_rep[!(l_rep %in% c("Negative_1","Negative_2"))]
phy_otu.clean.neg <- subset_samples(phy_otu.clean.neg, ReplicateID %in% l_noneg)
# 37 samples

## subset samples
# phy.clean.ss <- subset_samples(phy.clean, Label %in% lb)
# (phy.clean.ss)
# phy.clean.ss <- filter_taxa(phy.clean.ss, function(x) sum(x) != 0, TRUE)
# phy.clean.ss  ## 393
# sum(otu_table(phy.clean.ss)) # 1576881

### remove otus with sequence length
## what is the taxa? over 300
# over300 = c('1866097706fa91e752c8664faf43449f', '1b2f92e822b1d78946e428e4a0cb3c0f', '3561ce5acea02d8722df1647ac04134e', '380c415e118a20e871a9212093b89dcd', '44acd89e2a099c5838d390e7fd811fb4', '7142aab19a3913224dca238c45100219', '8227de12305c6fafbcf0aeb2c7ac0a35', 'ABRL02000548.13901.15510', 'ADWL01006363.13909.15531', 'CBQP010013872.5787.7604', 'LONC01000611.65580.67188', 'MTTA01000002.173539804.173541537', 'a3aa1c5831dbdac10601b422bf713873', 'ac88961e75ff2a36681aa818ef9cc4ee', 'bb35191255f4429dad23dad82575ea78', 'd77161ae99e2715c32f077131af656ea', 'df3998cea8301598580b90cd0e0a37b6', 'e00a18e93f328393803417e3be829423', 'e74c4dd0d5f55cbba155bd31da7b7f4e', 'f248cd73405b872671c04e6bec3a530e')
# 
# tax.over300 <-  tax_table(phy)[over300]
# tax.over300 <- data.frame(tax.over300)
# is.na(tax.over300$Order)
# tax.over300$Order
# phy.clean.ss <- pop_taxa(phy.clean.ss, over300 )
# phy.clean.ss  ## 390
# sum(otu_table(phy.clean)) # 1580210

## exclude taxa less than 5 sequences
phy_otu.clean.neg

otu_table(phy_otu.clean.neg) %>% rowSums() %>% sort()

v.less5 <- rownames(otu_table(phy_otu.clean.neg)[otu_table(phy_otu.clean.neg) %>% rowSums() < 5])
v.less10 <- rownames(otu_table(phy_otu.clean.neg)[otu_table(phy_otu.clean.neg) %>% rowSums() < 10])
v.less20 <- rownames(otu_table(phy_otu.clean.neg)[otu_table(phy_otu.clean.neg) %>% rowSums() < 20])
v.less30 <- rownames(otu_table(phy_otu.clean.neg)[otu_table(phy_otu.clean.neg) %>% rowSums() < 30])
v.less40 <- rownames(otu_table(phy_otu.clean.neg)[otu_table(phy_otu.clean.neg) %>% rowSums() < 40])
v.less50 <- rownames(otu_table(phy_otu.clean.neg)[otu_table(phy_otu.clean.neg) %>% rowSums() < 50])

length(v.less5)   ## 664
length(v.less10)  ## 1028
length(v.less20)  ## 1299
length(v.less30)  ## 1411
length(v.less40)  ## 1470
length(v.less50)  ## 1516

## execute
phy_otu  ##  2422 
phy_otu.clean.neg ## 1802  negative, unassigned taxa (kingdom, phy_otulum) removed.
# phy_otu.clean.ss   ## 390 otu
(phy_otu.clean.neg.5 <- pop_taxa(phy_otu.clean.neg, v.less5))   # 1138 
(phy_otu.clean.neg.10 <- pop_taxa(phy_otu.clean.neg, v.less10)) # 774  
(phy_otu.clean.neg.20 <- pop_taxa(phy_otu.clean.neg, v.less20)) # 503  
(phy_otu.clean.neg.30 <- pop_taxa(phy_otu.clean.neg, v.less30)) # 391  
(phy_otu.clean.neg.40 <- pop_taxa(phy_otu.clean.neg, v.less40)) # 332  
(phy_otu.clean.neg.50 <- pop_taxa(phy_otu.clean.neg, v.less50)) # 286 

sum(otu_table(phy_otu)) ## 170857 otu reads.
sum(otu_table(phy_otu.clean)) ## 168696
sum(otu_table(phy_otu.clean.neg)) ## 158624
sum(otu_table(phy_otu.clean.neg.5)) ## 156773
sum(otu_table(phy_otu.clean.neg.10)) ## 154355
sum(otu_table(phy_otu.clean.neg.20)) ## 150581

```

## plot relative abundance plot

```{r}
## average or sum replicates
# phy.m <-  merge_samples(phy.clean.neg, "Sample_name")
# tax_table(phy.m)

# we are going to plot replicates
l_rep <- c('Barneveld2_T0_R1', 'Barneveld2_T0_R2', 'Barneveld2_T0_R3', 'Barneveld2_T9_H0_R1', 'Barneveld2_T9_H0_R2', 'Barneveld2_T9_H0_R3', 'Barneveld2_T9_H0_CHL_R1', 'Barneveld2_T9_H0_CHL_R2', 'Barneveld2_T9_H0_CHL_R3', 'Barneveld2_T9_OH10_R1', 'Barneveld2_T9_OH10_R2', 'Barneveld2_T9_OH10_R3', 'Barneveld2_T9_OH10_CHL_R1', 'Barneveld2_T9_OH10_CHL_R2', 'Barneveld2_T9_OH10_CHL_R3', 'LaBaghWoods1_T0_R1', 'LaBaghWoods1_T0_R2', 'LaBaghWoods1_T0_R3', 'LaBaghWoods1_T9_H0_R1', 'LaBaghWoods1_T9_H0_R2', 'LaBaghWoods1_T9_H0_CHL_R1', 'LaBaghWoods1_T9_H0_CHL_R2', 'LaBaghWoods1_T9_OH8_R1', 'LaBaghWoods1_T9_OH8_R2', 'LaBaghWoods1_T9_OH8_CHL_R1', 'LaBaghWoods1_T9_OH8_CHL_R2', 'PMA_Barneveld2_T0_R1', 'PMA_Barneveld2_T0_R2', 'PMA_Barneveld2_T9_H0_R1', 'PMA_Barneveld2_T9_H0_R2', 'PMA_Barneveld2_T9_H0_CHL_R1', 'PMA_Barneveld2_T9_H0_CHL_R2', 'PMA_Barneveld2_T9_OH10_R1', 'PMA_Barneveld2_T9_OH10_R2', 'PMA_Barneveld2_T9_OH10_CHL_R1', 'PMA_Barneveld2_T9_OH10_CHL_R2', 'Positive')

## let's get phylum sir
df.phylum <- phy_otu.clean.neg %>%
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  psmelt()
df.phylum$ReplicateID <- factor(df.phylum$ReplicateID, levels = l_rep)

library(forcats) 
df.phylum %<>% mutate(Phylum = fct_explicit_na(Phylum, na_level = "unidentified"))
unique(df.phylum$Phylum)

levels(df.phylum$Phylum)
levels(df.phylum$Phylum) = c(levels(df.phylum$Phylum), 'Low abundance')

# we need to group by samples
df.phylum.rel <- df.phylum %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

# df.phylum.rel[df.phylum.rel$RelAbundance < 5,]$Phylum <- 'Low abundance'

### let's test
ord <- df.phylum.rel %>% group_by(Phylum) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Phylum
rev(tail(vec,10))
vec
# x = factor(x,levels(x)[c(4,5,1:3)])
# levels(vec) <- c(levels(vec), 'Low abundance')
# vec
# length(vec)
# levels(vec) <- c(length(vec),1:length(vec)-1)
# vec

vec
# rest <- vec[!vec %in% c('Low abundance')]
# vec <- factor(vec,levels =  )
rev(vec)
# New

# vec <- c('Low abundance','unidentified','Acidobacteria','Bacteroidetes','Firmicutes','Actinobacteria','Proteobacteria')


df.phylum.rel$Phylum <- factor(df.phylum.rel$Phylum, levels = vec)

## relative abundance with less than 0.5% in sample was labeled 'low abundance'

## plot relative abundance
ggplot(df.phylum.rel, aes(x=ReplicateID, y = RelAbundance, fill = Phylum)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection) +
  
  xlab('')+
  ylab("Relative Abundance (%) \n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(nrow = 4,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())


##### now class
## let's get class sir
df.class <- phy_otu.clean.neg %>%
  tax_glom(taxrank = "Class", NArm = FALSE) %>% 
  psmelt()
df.class$ReplicateID <- factor(df.class$ReplicateID, levels = l_rep)

library(forcats) 
df.class %<>% mutate(Class = fct_explicit_na(Class, na_level = "unidentified"))
unique(df.class$Class)

levels(df.class$Class)
levels(df.class$Class) = c(levels(df.class$Class), 'Low abundance')

# we need to group by samples
df.class.rel <- df.class %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.class.rel[df.class.rel$RelAbundance < 1,]$Class <- 'Low abundance'

### let's test
ord <- df.class.rel %>% group_by(Class) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Class
# rev(tail(vec,10))
# vec
# write.table(vec,'vec.txt', quote=FALSE, sep='\t', row.names = F, col.names = F)

df.class.rel$Class <- factor(df.class.rel$Class, levels = vec)

## 2. plot relative abundance
ggplot(df.class.rel, aes(x=ReplicateID, y = RelAbundance, fill = Class)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(nrow = 3,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())

##### now Order

df.order <- phy_otu.clean.neg %>%
  tax_glom(taxrank = "Order", NArm = FALSE) %>% 
  psmelt()
df.order$ReplicateID <- factor(df.order$ReplicateID, levels = l_rep)

library(forcats) 
df.order %<>% mutate(Order = fct_explicit_na(Order, na_level = "unidentified"))
unique(df.order$Order)

levels(df.order$Order) = c(levels(df.order$Order), 'Low abundance')

# we need to group by samples
df.order.rel <- df.order %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.order.rel[df.order.rel$RelAbundance < 0.3,]$Order <- 'Low abundance'

### let's test
ord <- df.order.rel %>% group_by(Order) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Order
# rev(tail(vec,10))
# vec
# write.table(vec,'vec.txt', quote=FALSE, sep='\t', row.names = F, col.names = F)

df.order.rel$Order <- factor(df.order.rel$Order, levels = vec)
length(vec)
## 2. plot relative abundance
ggplot(df.order.rel, aes(x=ReplicateID, y = RelAbundance, fill = Order)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(ncol = 6,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())


##### now family
df.family <- phy_otu.clean.neg %>%
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  psmelt()
df.family$ReplicateID <- factor(df.family$ReplicateID, levels = l_rep)

library(forcats) 
df.family %<>% mutate(Family = fct_explicit_na(Family, na_level = "unidentified"))
unique(df.family$Family)

levels(df.family$Family) = c(levels(df.family$Family), 'Low abundance')

# we need to group by samples
df.family.rel <- df.family %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.family.rel[df.family.rel$RelAbundance < 0.3,]$Family <- 'Low abundance'


### let's test
ord <- df.family.rel %>% group_by(Family) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- ord$Family
length(vec)
# rev(tail(vec,10))
# vec
# length(vec)
# write.table(vec,'vec.txt', quote=FALSE, sep='\t', row.names = F, col.names = F)
df.family.rel$Family <- factor(df.family.rel$Family, levels = vec)

## 2. plot relative abundance
ggplot(df.family.rel, aes(x=ReplicateID, y = RelAbundance, fill = Family)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection2) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(ncol = 7,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())



##### now Genus
### let's get phylum sir
# prune out phyla below 2% in each sample
df.genus <- phy_otu.clean.neg %>%
  tax_glom(taxrank = "Genus", NArm = FALSE) %>% 
  psmelt()
df.genus$ReplicateID <- factor(df.genus$ReplicateID, levels = l_rep)

# library(dplyr)
# library(forcats) 
# library(magrittr)
df.genus %<>% mutate(Genus = fct_explicit_na(Genus, na_level = "unidentified"))
unique(df.genus$Genus)

levels(df.genus$Genus) = c(levels(df.genus$Genus), 'Low abundance')

# we need to group by samples
df.genus.rel <- df.genus %>%  
  group_by(Sample) %>%                         # Filter out at absolute read of 20       
  mutate(RelAbundance = Abundance*100/sum(Abundance))  # Transform to rel. abundance

df.genus.rel[df.genus.rel$RelAbundance < 0.3,]$Genus <- 'Low abundance'

### let's test

gen.ord <- df.genus.rel %>% group_by(Genus) %>% summarise(Abundance = sum(Abundance)) %>% arrange(Abundance)
vec <- gen.ord$Genus
length(vec)
# rev(tail(vec.gen,10))
# vec.gen
# write.xlsx(rev(vec.gen), 'bacteria_genus_list.xlsx')

# let's make list of genus starting from kingdom
df.genus.rel$Genus <- factor(df.genus.rel$Genus, levels = vec)

genus_color <- c(
  'gray','black',"#F92A82", "#ED7B84", "#7EB77F", 
  "#DEC4A1", "#E5D1D0", '#0E8482', '#C9DAEA', '#337357', 
  '#95C623', '#E55812', '#04471C', '#F2D7EE', '#D3BCC0', 
  '#A5668B', '#69306D', '#0E103D', '#1A535C', '#4ECDC4', 
  '#F7FFF7', '#FF6B6B', '#FFE66D', '#6699CC', '#FFF275', 
  '#FF8C42', '#FF3C38', '#A23E48', '#000000', '#CF5C36', 
  '#EEE5E9', '#7C7C7C', '#EFC88B', '#2E5266', '#6E8898', 
  
  '#7ac864', '#ff7473', '#00e2ff', '#8836ff', '#732b90',
  '#e72fff', '#006eff', '#e7eb1c','#00ffe0','#ffc680',
  '#ABD0BC','#519eff', '#ffaa00','#00a300','#b23b00',
  '#ecb0e7','#ff5a00' ,'#8ee7e7'
)
## 2. plot relative abundance
ggplot(df.genus.rel, aes(x=ReplicateID, y = RelAbundance, fill = Genus)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection2) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  mytheme_2d+
  guides(fill = guide_legend(ncol = 7,reverse = T))+
  theme(legend.position="bottom") +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  scale_y_continuous(breaks=seq(0,100,10))+
  scale_x_discrete(expand = c(0, 0)) +
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(), panel.background=element_blank(),panel.border=element_blank(), plot.background=element_blank())


# plot only the positive
df.genus.rel.positive = df.genus.rel %>% filter(ReplicateID == "Positive")

ggplot(df.genus.rel.positive, aes(x=ReplicateID, y = RelAbundance, fill = Genus)) + 
  geom_bar(stat="identity", width = 0.8, position = 'stack', colour="black") +
  #scale_fill_discrete() +
  scale_fill_manual(values = my_color_collection2) +
  
  xlab('')+
  ylab("Relative Abundance (%)\n") +
  #ggtitle("Phylum Community Composition by Sample \n") +
  ## adjust positions
  guides(fill = guide_legend(ncol = 2,reverse = T)) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face='bold')) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face='bold')) + 
  theme(axis.text.x = element_text(size=12, angle = 50, hjust = 1, vjust=1.1, face='bold',color='black'), axis.ticks.x=element_blank())+
  theme(axis.text.y = element_text(size=15, face='bold',color='black'))+
  mytheme_2d +
  theme(legend.position="bottom")

```


## Plot OTU rank abundance

```{r}
## convert OTU table into relative abundance
# how about using phy_otu.clean.neg.10
otu_rel <- otu_table(phy_otu.clean.neg.10) # remove otu that has less than 10 reads total.
class(otu_rel)

hist(rowSums(otu_rel), breaks = 60)
table(rowSums(otu_rel))
sort(rowSums(otu_rel)) %>% tail()

mat_rel <- apply(otu_rel,2, function(x) x/sum(x))
dim(mat_rel) # 1903 otu
colSums(mat_rel)

## Reorder the rows and change rownames
sort(rowSums(otu_rel), decreasing = T)
names(sort(rowSums(otu_rel), decreasing = T))

mat_rel2 <- mat_rel[names(sort(rowSums(otu_rel), decreasing = T)),,drop=F]
rownames(mat_rel2) <- paste0("otu", sprintf("%04d", 1:dim(mat_rel2)[1]))

mat_rel2
colSums(mat_rel2)
dim(mat_rel2)

## rank matrix per column
library(matrixStats)

## how to calculate ranks
x <- cbind(a=c(11,22,44),
       b=c(21,12,NA),
       c=c(35,66,12))
x
# apply(x, 2, rank)
apply(-x, 2, rank)

# make 0 NA
any(is.na(mat_rel2)) # no NA in the data
mat_rel2[mat_rel2 == 0] <- NA
dim(mat_rel2)
sum(is.na(mat_rel2)) # 24103 NAs

mat_rank <- apply(-mat_rel2, 2, rank)
any(is.na(mat_rank)) # there should be NA
dim(mat_rank)
# change back to NA, which are supposed to be NA
mat_rank[is.na(mat_rel2)] <- NA
any(is.na(mat_rank))
sum(is.na(mat_rank)) # 24103 NAs

colMaxs(mat_rank, na.rm =T)
colMins(mat_rank, na.rm =T)

mat_rank[,1]
table(mat_rank[,1])
table(mat_rank[,2])
table(mat_rank[,3])
table(mat_rank[,4]) %>% tail()

mat_rank

## plot rank
sub_mat_rank <- mat_rank[,1:2]
mlt_rank <- melt(sub_mat_rank)
colnames(mlt_rank) <- c("OTU", "Sample", "Rank")
head(mlt_rank)
class(mlt_rank)

## get the rank abundance aligned to SampleA1
vec_otu_sampleA1 <- mlt_rank %>% filter(Sample == "SampleA1") %>% arrange(Rank) %>% select(OTU)
head(vec_otu_sampleA1)
vec_otu_sampleA1 <- vec_otu_sampleA1[,1]
mlt_rank$OTU <- factor(mlt_rank$OTU, levels = vec_otu_sampleA1)

ggplot(mlt_rank, aes(x = OTU, y = Rank)) +
  geom_line(aes(col=Sample)) +
  geom_point(aes(col=Sample)) +
  ylab("Rank \n") +
  xlab(paste0("\n OTU ID")) +
  mytheme_2d +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
  # scale_y_continuous(expand = c(0, 0))

## Okay this is a good demonstration of the data


```

## Let's begin the proper analysis

```{r}
## connect metadata to the otu table

phy_otu.clean.neg.10
df_meta <- sample_data(phy_otu.clean.neg.10)

## melt the total rank matrix into dataframe, then attach the metadata
mat_rank
mlt_rank <- melt(mat_rank)
colnames(mlt_rank) <- c("OTU", "Sample", "Rank")
head(mlt_rank)

dim(mlt_rank) #70411 3
mlt_rank_meta <- mlt_rank %>% left_join(df_meta, by=c("Sample"="SampleID"))
dim(mlt_rank_meta) #70411 14
colnames(mlt_rank_meta)

# replicate check - start with 1 sample
mlt_rank_meta %>% select(Sample_name) %>% unique()
mlt_B2_T0 <- mlt_rank_meta %>% filter(Sample_name == "Barneveld2_T0") 
dim(mlt_B2_T0)
head(mlt_B2_T0)


# plot
## get the rank abundance aligned to Barneveld2_T0
vec_otu_B2T0 <- mlt_B2_T0 %>% filter(Replicate == 1) %>% arrange(Rank) %>% select(OTU)
dim(vec_otu_B2T0)
head(vec_otu_B2T0)
vec_otu_B2T0 <- vec_otu_B2T0[,1]
mlt_B2_T0$OTU <- factor(mlt_B2_T0$OTU, levels = vec_otu_B2T0)

mlt_B2_T0$Replicate <- factor(mlt_B2_T0$Replicate)　　　

ggplot(mlt_B2_T0 %>% filter(Replicate ==1), aes(x = OTU, y = Rank)) +
  # geom_line(aes(col=Sample)) +
  geom_point(aes(col=Replicate)) +
  ggtitle("Barneveld2_T0") +
  ylab("Rank \n") +
  xlab("\n OTU ID") +
  mytheme_2d +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())

# what is happening here?
ggplot(mlt_B2_T0, aes(x = OTU, y = Rank)) +
  # geom_line(aes(col=Sample)) +
  geom_point(aes(col=Replicate)) +
  ggtitle("Barneveld2_T0") +
  ylab("Rank \n") +
  xlab("\n OTU ID") +
  mytheme_2d +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
  # scale_y_continuous(expand = c(0, 0))

# upto rank 500
# 
# mlt_B2_T0_rank500 <- mlt_B2_T0 %>% filter(Rank < 500)
# vec_B2_T0_rank500 <- mlt_B2_T0_rank500 %>% filter(Replicate == 1) %>% arrange(Rank) %>% select(OTU)
# head(vec_B2_T0_rank500)
# vec_B2_T0_rank500 <- vec_B2_T0_rank500[,1]
# mlt_B2_T0_rank500$OTU <- factor(mlt_B2_T0_rank500$OTU, levels = vec_B2_T0_rank500)
# 
# 
# ggplot(mlt_B2_T0_rank500, aes(x = OTU, y = Rank)) +
#   # geom_line(aes(col=Replicate)) +
#   geom_point(aes(col=Replicate)) +
#   ylab("Rank \n") +
#   xlab("\n OTU ID") +
#   ggtitle("Barneveld2_T0 zoom-in rank500") +
#   mytheme_2d +
#   theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())

## plot across different samples, align otu to replicate 1
l_sample_names <- mlt_rank_meta %>% select(Sample_name) %>% unique()
l_sample_names <- l_sample_names[,1]
l_sample_names <- l_sample_names[-length(l_sample_names)] # remove positive sample

plot_rank_otu <- function(mlt_rank_meta, Sample_name_i = "Barneveld2_T0", Replicate_i = 1){
  mlt_B2_T0 <- mlt_rank_meta %>% filter(Sample_name == Sample_name_i) 
  dim(mlt_B2_T0)
  head(mlt_B2_T0)
  
  ## get the rank abundance aligned to Barneveld2_T0
  vec_otu_B2T0 <- mlt_B2_T0 %>% filter(Replicate == Replicate_i) %>% arrange(Rank) %>% select(OTU)
  dim(vec_otu_B2T0)
  head(vec_otu_B2T0)
  vec_otu_B2T0 <- vec_otu_B2T0[,1]
  mlt_B2_T0$OTU <- factor(mlt_B2_T0$OTU, levels = vec_otu_B2T0)
  mlt_B2_T0$Replicate <- factor(mlt_B2_T0$Replicate)　　　
  
  ggplot(mlt_B2_T0, aes(x = OTU, y = Rank)) +
    # geom_line(aes(col=Sample)) +
    geom_point(aes(col=Replicate)) +
    ggtitle(paste0(Sample_name_i)) +
    ylab("Rank") +
    xlab("OTU ID") +
    mytheme_2d +
    theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
    # scale_y_continuous(expand = c(0, 0))
}

# plot in 1 sheet
l_plot <- list()
for (i in 1:length(l_sample_names)) {
    print(i)
    l_plot[[i]] <- plot_rank_otu(mlt_rank_meta, l_sample_names[i], Replicate_i=1)
}

library(gridExtra)
grid.arrange(grobs = l_plot, ncol = 5)

l_plot <- list()
for (i in 1:length(l_sample_names)) {
    print(i)
    l_plot[[i]] <- plot_rank_otu(mlt_rank_meta, l_sample_names[i], Replicate_i=2)
}

library(gridExtra)
grid.arrange(grobs = l_plot, ncol = 5)

l_plot <- list()
for (i in 1:length(l_sample_names)) {
    print(i)
    l_plot[[i]] <- plot_rank_otu(mlt_rank_meta, l_sample_names[i], Replicate_i=3)
}

library(gridExtra)
grid.arrange(grobs = l_plot, ncol = 5)
## plot abundance - rank plot to see how much abundance is rank 500
mat_rel2
mlt_rank_meta
colnames(mlt_rank_meta)

mlt_rel2 <- melt(mat_rel2)
colnames(mlt_rel2) <- c("OTU", "Sample", "Rel_abundance")
head(mlt_rel2)

# join two dataframes
mlt_rank_rel <- mlt_rank_meta %>% left_join(mlt_rel2, by = c("OTU"="OTU", "Sample"="Sample"))
colnames(mlt_rank_rel)
# plot Rank (x) - relative abundance (y) plots
l_sample_names

mlt_rank_rel_B2_T0 <- mlt_rank_rel %>% filter(Sample_name == "Barneveld2_T0")
mlt_rank_rel_B2_T0$Replicate <- factor(mlt_rank_rel_B2_T0$Replicate)

dim(mlt_rank_rel_B2_T0)

# what is happening here?
mlt_rank_rel_B2_T0 %>% select(Rank, Rel_abundance) # oh no... got ranked as well...


dim(na.omit(mlt_rank_rel_B2_T0))
mlt_rank_rel_B2_T0_noNA <- na.omit(mlt_rank_rel_B2_T0)

ggplot(mlt_rank_rel_B2_T0_noNA, aes(x = Rank, y = Rel_abundance)) +
  geom_line(aes(col=Replicate)) +
  geom_point(aes(col=Replicate)) +
  ggtitle(paste0("Barneveld2_T0")) +
  ylab("Relative abundance") +
  xlab("Rank") +
  mytheme_2d

mlt_rank_rel_B2_T0_noNA %>% group_by(Replicate) %>% summarize(Total = sum(Rel_abundance))

## plot Rel_abundance - rank plots for all samples
plot_rank_abundance <- function(mlt_rank_rel, Sample_name_i){
  mlt_rank_rel_B2_T0 <- mlt_rank_rel %>% filter(Sample_name == Sample_name_i)
  
  mlt_rank_rel_B2_T0$Replicate <- factor(mlt_rank_rel_B2_T0$Replicate)
  
  mlt_rank_rel_B2_T0_noNA <- na.omit(mlt_rank_rel_B2_T0)
  df_mlt_plot <- mlt_rank_rel_B2_T0_noNA %>% select(Sample_name, Replicate, Rank, Rel_abundance) %>% ungroup()
  colnames(df_mlt_plot)
  class(df_mlt_plot)
  # print("we are good")
  
  ggplot(df_mlt_plot, aes(x=Rank, y=Rel_abundance)) +
    geom_point(aes(col=Replicate)) +
    geom_line(aes(col=Replicate)) +
    ggtitle(paste0(Sample_name_i)) +
    ylab("Relative abundance") +
    xlab("Rank") +
    mytheme_2d
}

# plot in 1 sheet
l_plot2 <- list()
for (i in 1:length(l_sample_names)) {
  print(i)
  # print(l_sample_names[i])
  l_plot2[[i]] <- plot_rank_abundance(mlt_rank_rel, l_sample_names[i])
}

library(gridExtra)
grid.arrange(grobs = l_plot2, ncol = 5)

```


## Let's visualize the overlap between replicates

```{r}
phy_otu.clean.neg.10

df_subset <- psmelt(phy_otu.clean.neg.10) %>%  select(OTU, Sample, Replicate, Abundance) %>% group_by(Sample) %>%  filter(Abundance > 0) %>% arrange(Sample)

dim(df_subset)

df_otu_1 <- df_subset %>% filter(Replicate == 1) %>% select(OTU) %>% unique()
df_otu_1$OTU

venn_each_sample <- function(phy_otu.clean.neg.10){
  library(VennDiagram)
  df_subset <- psmelt(phy_otu.clean.neg.10) %>%  select(OTU, Sample, Replicate, Abundance) %>% group_by(Sample) %>%  filter(Abundance > 0) %>% arrange(Sample)
  df_otu_1 <- df_subset %>% filter(Replicate == 1) %>% select(OTU) %>% unique()
  vec.f <- df_otu_1$OTU
  df_otu_2 <- df_subset %>% filter(Replicate == 2) %>% select(OTU) %>% unique()
  vec.s <- df_otu_2$OTU
  df_otu_3 <- df_subset %>% filter(Replicate == 3) %>% select(OTU) %>% unique()
  vec.t <- df_otu_3$OTU
  n12 <- intersect(vec.f,vec.s)
  n23 <- intersect(vec.s,vec.t)
  n13 <- intersect(vec.f,vec.t)
  n123 <- intersect(intersect(vec.f,vec.s),vec.t)
  
  my.list <- list(first = vec.f,second = vec.s, third=vec.t, n12=n12, n23=n23,n13=n13,n123=n123)
  my.list
  draw.triple.venn(area1 = length(my.list$first),
                   area2 = length(my.list$second),
                   area3 = length(my.list$third),
                   n12 = length(my.list$n12),
                   n23 = length(my.list$n23),
                   n13 = length(my.list$n13),
                   n123 = length(my.list$n123),
                   #category = c("first","second",'third'),
                   fill = c("#FE9696",'#96FE96',"#9696FE"),
                   euler.d = F,
                   scaled = F,
                   ext.text = T,
                   ext.percent = c(0.1,0.1,0.1),
                   ext.length = 1,
                   label.col = rep("gray10",7),
                   lwd = 0,
                   cex = 1,
                   fontface = rep("plain",7),
                   fontfamily = rep("sans",7), 
                   # cat.cex = 1.5,
                   # cat.fontface = rep("bold",3),
                   # cat.fontfamily = rep("sans",3),
                   # #cat.pos = c(0, 0,0),
                   print.mode = c("percent","raw")
  )
}


l_venn <- list()
for (i in 1:length(l_sample_names)){
  print(i)
  j <- l_sample_names[i]
  phy.ss1 <- subset_samples(phy_otu.clean.neg.10, Sample_name == j)
  l_venn[[i]] <- grid.arrange(gTree(children=venn_each_sample(phy.ss1)), top = paste0(j))
}
length(l_venn)

grid.arrange(grobs = l_venn, ncol=5)

```

## How many strains take up 80% of reads of each sample

```{r}
# install.packages("ineq")
library(ineq)


df_asv_total <- psmelt(phy.clean.neg.10) %>% group_by(OTU) %>% summarize(Total =sum(Abundance))
v_asv_total <- df_asv_total$Total

df_otu_total <-psmelt(phy_otu.clean.neg.10) %>% group_by(OTU) %>% summarize(Total =sum(Abundance))
v_otu_total <- df_otu_total$Total

# 1. ASV
par(font.axis = 2, font.lab=2)
plot(Lc(v_asv_total), col='navy',lwd=2, xlab='Cumulative percentage of ASVs from the lowest to highest abundance %', ylab='Cumulative share from total abundance %', main='Cumulative share of abundance by ASVs (Lorenz curve)')
ineq(v_asv_total,type="Gini") # returns 0.7971297

lc.asv <- Lc(v_asv_total)
lc.asv$L[700]

# 2. OTU
par(font.axis = 2, font.lab=2)
plot(Lc(v_otu_total), col='red4',lwd=2, xlab='Cumulative percentage of OTUs from the lowest to highest abundance %', ylab='Cumulative share from total abundance %', main='Cumulative share of abundance by OTUs (Lorenz curve)')
ineq(v_otu_total,type="Gini") # returns 0.8109097

## let's construct added graphs
plot(Lc(v_asv_total), col='navy',lwd=3, xlab='Cumulative percentage of strains from the lowest to highest abundance %', ylab='Cumulative share of total abundance %', main='Cumulative share of abundance by strains (Lorenz curve)')
lines(Lc(v_otu_total), col='red3',lwd=3)
legend('topleft',pch =c(15), legend=c('ASV','OTU'),
       col=c('navy','red3'), bty="n")


```









